cmd/fpga_admissionwebhook/patcher_test.go

```diff
@@ -0,0 +1,353 @@
+						"cpu": resource.MustParse("1"),
+					},
+					Requests: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+						"cpu": resource.MustParse("1"),
+					},
+				},
+			},
+			regionMap: map[string]string{
+				"arria10": "ce48969398f05f33946d560708be108a",
+			},
+			afMap: map[string]string{
+				"arria10-nlb0": "d8424dc4a4a3c413f89e433683f9040b",
+			},
+			expectedOps: 3,
+		},
+		{
+			name: "More than one FPGA in Limits",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+						"fpga.intel.com/arria10-nlb3": resource.MustParse("1"),
+					},
+				},
+			},
+			regionMap: map[string]string{
+				"arria10": "ce48969398f05f33946d560708be108a",
+			},
+			afMap: map[string]string{
+				"arria10-nlb0": "d8424dc4a4a3c413f89e433683f9040b",
+				"arria10-nlb3": "f7df405cbd7acf7222f144b0b93acd18",
+			},
+			expectedErr: true,
+		},
+		{
+			name: "More than one FPGA in Requests",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+						"fpga.intel.com/arria10-nlb3": resource.MustParse("1"),
+					},
+				},
+			},
+			regionMap: map[string]string{
+				"arria10": "ce48969398f05f33946d560708be108a",
+			},
+			afMap: map[string]string{
+				"arria10-nlb0": "d8424dc4a4a3c413f89e433683f9040b",
+				"arria10-nlb3": "f7df405cbd7acf7222f144b0b93acd18",
+			},
+			expectedErr: true,
+		},
+		{
+			name: "Unknown FPGA model in Requests",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						"fpga.intel.com/unknown-nlb0": resource.MustParse("1"),
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "Unknown AFU in Requests",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						"fpga.intel.com/arria10-unknown": resource.MustParse("1"),
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "Unknown FPGA model in Limitss",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"fpga.intel.com/unknown-nlb0": resource.MustParse("1"),
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "Unknown AFU in Limits",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"fpga.intel.com/arria10-unknown": resource.MustParse("1"),
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "Wrong ENV",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+					},
+				},
+				Env: []corev1.EnvVar{
+					{
+						Name:  "FPGA_REGION",
```

Comment:
test for FPGA_AFU in user provided spec?

Reply:
This should be a test case for `getEnvVars()` function, not `getPatchOpsOrchestrated()`. I'll add it there.